PLMS.BLL/ServicesImplementation/TaskService.cs

```diff
@@ -1,206 +1,338 @@
-                (filters.StatusIds == null || filters.StatusIds.Contains(t.StatusId)),
-                q => q.OrderByDescending(t => t.TaskComments.Select(tc => tc.CreatedAt)),
-                "TaskComments,Goal");
-
-            if (!tasks.Any())
-            {
-                return Enumerable.Empty<TaskDto>();
-            }
+            var tasks = await GetFilteredTasksAsync(filters);
 
-            if (tasks.Count() < filters.ItemsPerPageCount)
+            if (tasks == null || !tasks.Any())
             {
-                filters.ItemsPerPageCount = tasks.Count();
+                return Enumerable.Empty<TaskShortWithCommentsDto>();
             }
 
-            var results = tasks.Take(filters.ItemsPerPageCount).Select(t =>
-            new TaskDto
+            var results = tasks.Select(t => new TaskShortWithCommentsDto
             {
                 Title = t.Title,
-                GoalTitle = t.Goal.Title,
-                Comments = t.TaskComments.Select(tс => new TaskCommentDto 
+                GoalTitle = t.Goal?.Title ?? string.Empty,
+                TaskComments = t.TaskComments?.Select(tс => new TaskCommentDto 
                 { 
                     Comment = tс.Comment, 
-                    CreatedAt = tс.CreatedAt 
-                }).ToArray()
-
+                    CreatedAt = tс.CreatedAt
+                })?
+                .ToArray() ?? Array.Empty<TaskCommentDto>(),
+                CreatedAt = t.CreatedAt
             }).ToArray();
 
             return results;
         }
 
-        public async Task<TaskDto> GetTaskByIdAsync(int id)
+        public async Task<GetTaskDto> GetTaskByIdAsync(int id, string userId)
         {
-            var task = await _taskRepository.GetByIdAsync(id);
+            var taskRepository = _unitOfWork.GetRepository<Task>();
+            var goalRepository = _unitOfWork.GetRepository<Goal>();
+            var task = await taskRepository.GetByIdAsync(id);
+            if (task == null)
+            {
+                return new GetTaskDto();
+            }
 
-            //ToDo: all null checking for entire task object as well as for some fields like Priority, Status, Goal, etc
+            var goal = await goalRepository.GetByIdAsync(task.GoalId);
+            if (goal.UserId != userId)
+            {
+                throw new UnauthorizedAccessException("This user is not unauthorized to access this task");
+            }
 
-            TaskDto taskDto = new()
+            var taskDto = new GetTaskDto()
             {
+                Id = task.Id,
                 Title = task.Title,
                 Description = task.Description,
-                GoalTitle = task.Goal.Title,
-                Comments = task.TaskComments.Select(
+                GoalId = task.GoalId,
+                CreatedAt = task.CreatedAt,
+                DueDate = task.DueDate,
+                PriorityId = task.PriorityId,
+                StatusId = task.StatusId
+            };
```

Comment:
This can be done by AutoMapper, just make sure it will not cause unexpected lazy loading.
Same for all Get methods of the TaskService